filter_xml/data_processor.py

```diff
@@ -154,3 +155,47 @@ def valid_production_unit(self, restaurant: dict) -> bool:
         p-number and a CVR number.
         """
         return self._has_pnr(restaurant) and self._has_cvr(restaurant)
+
+    def create_diff(self, res: list) -> tuple:
+        current_db = self._outputter.get()
+        current_ids = {x['name_seq_nr'] for x in current_db}
+        res_ids = {x['name_seq_nr'] for x in res}
+
+        res_by_key = {x['name_seq_nr']: x for x in res}
+        current_by_key = {x['name_seq_nr']: x for x in current_db}
```

Comment:
Is it possible to use only `res_by_key` and `current_by_key` to avoid constructing all 4 data structures?

Reply:
and then use `.keys()`

Reply:
It is, but complexity-wise it would be the same.